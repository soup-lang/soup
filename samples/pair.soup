func make_pair(x, y)
    func selector(select)
        return select(x, y)
    return selector

func first(x, y)
    return x

func second(x, y)
    return y

m: 10
x: make_pair(m, 6)
console.log(x(first))
console.log(x(second))

y: make_pair(false, true)
console.log(y(first))
console.log(y(second))
