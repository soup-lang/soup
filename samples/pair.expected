10
6
false
true
